src/azul/plugins/repository/tdr_anvil/__init__.py

```diff
@@ -457,14 +457,20 @@ def _supplementary_bundle(self, bundle_fqid: TDRAnvilBundleFQID) -> TDRAnvilBund
     def _duos_bundle(self, bundle_fqid: TDRAnvilBundleFQID) -> TDRAnvilBundle:
         assert not bundle_fqid.is_batched, bundle_fqid
         duos_info = self.tdr.get_duos(bundle_fqid.source)
```

Comment:
As far as I can tell, get_duos() already uses the ID and could return it. If that is not the case, I would like to understand why. If it is, that method should return a tuple of the ID and the info, after asserting that the ID used to look up the info, and the ID embedded in it are equal.